torch/_refs/__init__.py

```diff
@@ -1280,8 +1280,8 @@ def isclose(
     # If the values of the integer tensors cannot be exactly represented
     # by the default scalar type then this may cause an incorrect result.
     if not utils.is_float_dtype(a.dtype) and not utils.is_complex_dtype(a.dtype):
-        a = prims.convert_element_type(a, torch.get_default_dtype())
-        b = prims.convert_element_type(b, torch.get_default_dtype())
+        a = _maybe_convert_to_dtype(a, torch.get_default_dtype())
```

Comment:
In this case the conversion will always occur since the default dtype is always a floating point type

Reply:
Hi, @mruberry! I checked into the functions your point completely makes sense to me. Thanks for pointing it out :)